### ISSUE TITLE
[Bug Report] ttnn.add with int32 and broadcasting gives wrong results

### ISSUE THREAD

**Issue by sgligorijevicTT** [tenstorrent/tt-metal#20582]
**Describe the bug**

**To Reproduce**
1. Run the following code:
```

import torch

import ttnn

x = torch.ones(1,1, dtype=torch.int32)
temp = torch.zeros(1,6, dtype=torch.int32)

with ttnn.manage_device(0) as dev:
    ttnn_x = ttnn.from_torch(x, dtype=ttnn.int32, layout=ttnn.TILE_LAYOUT, device=dev)
    ttnn_temp = ttnn.from_torch(temp, dtype=ttnn.int32, layout=ttnn.TILE_LAYOUT, device=dev)
    ttnn_y = ttnn.add(ttnn_x, ttnn_temp)
    print(ttnn_y)
    y = ttnn.to_torch(ttnn_y)

print(y)

```
2. Observe that the output is `TorchTensor([[0, 0, 0, 0, 0, 0]], dtype=torch.int32)`

**Expected behavior**
Tensor of all ones is printed.

**Please complete the following environment information:**
- OS: Ubuntu 22.04, default ird image
- Version of software (eg. commit): d686ad1407372f1cb854b760a0025ebb19e00bb8

**Comment by mrakitaTT** [comment]
Many of our models in Forge are failing with low accuracy, and this is the first suspect that we have found during investigation, so putting P1 priority.

@eyonland please let me know if there is already an issue tracking this. I suppose that [binary_ng](https://github.com/tenstorrent/tt-metal/issues/16147) will solve this, but I think it is important to track all the separate test cases that fail so we can confirm they are going to be fixed with binary_ng, and to confirm that they are indeed fixed once binary_ng lands.

**Comment by KalaivaniMCW** [comment]
@sgligorijevicTT  
please use `use_legacy=False` if inputs require broadcasting - this will switch to binary_ng
try  `ttnn.add(ttnn_x, ttnn_temp, use_legacy=False)`

**Comment by KalaivaniMCW** [comment]
> [@sgligorijevicTT](https://github.com/sgligorijevicTT) please use `use_legacy=False` if inputs require broadcasting - this will switch to binary_ng try `ttnn.add(ttnn_x, ttnn_temp, use_legacy=False)`

CC : @mrakitaTT

**Comment by KalaivaniMCW** [comment]
<img width="979" alt="Image" src="https://github.com/user-attachments/assets/fe632677-beb6-4b33-a921-3e5b5b9f86b8" />

**Comment by mrakitaTT** [comment]
Thank you @KalaivaniMCW, we've tried setting the `use_legacy=False` and it fixed this issue, but then it caused some other models that worked fine to fail with accuracy issues. It seems to me that `binary_ng` is not yet stable enough. We will open new issues when we locate the ttnn ops that are causing accuracy failures with `use_legacy=False`.

I would not yet close this one until `binary_ng` is enabled by default, because we cannot use `use_legacy=False` on our side because of these new issues it causes.

**Comment by cmaryanTT** [comment]
@mrakitaTT - yes, please open issues on accuracy problems. We should be superior on most functional issues in binary_ng, so anything that deviates from that would be worth looking at promptly.

**Comment by KalaivaniMCW** [comment]
@mrakitaTT - sure. While creating a new issue, please mention the ttnn ops that are causing accuracy failures, with the tested tensor shape and datatype info. If specific set of data is used for testing then attaching that input csv file would also help.

**Comment by mrakitaTT** [comment]
@cmaryanTT @KalaivaniMCW sorry for delay, we had some troubles (long story) isolating the repro case for models that failed with `use_legacy=false`. In the meantime after latest metal uplift those issues seem to be fixed, so now we will re-run all model tests to verify. I'll let you know if we find anything new.

**Comment by KalaivaniMCW** [comment]
The test now passes, it works with the default behaviour of the OP ever since `binary_ng` was switched on by default

<img width="1227" alt="Image" src="https://github.com/user-attachments/assets/af6cfc36-877b-4493-a8ef-b275b2189c8b" />

**Comment by KalaivaniMCW** [comment]
Note: if broadcasting is needed on dimension H (dim = -2), then `use_legacy=false` flag needs to specified explicitly
